classes/sso/user/mumie_user_service.php

```diff
@@ -0,0 +1,99 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * This class provides service functionalities for mumie_users
+ *
+ * @package auth_mumie
+ * @copyright  2017-2023 integral-learning GmbH (https://www.integral-learning.de/)
+ * @author Tobias Goltz (tobias.goltz@integral-learning.de)
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+namespace auth_mumie\user;
+
+defined('MOODLE_INTERNAL') || die();
+
+use auth_mumie\hash\hashing_service;
+use auth_mumie\hash\mumie_id_hash;
+
+require_once($CFG->dirroot . '/auth/mumie/classes/sso/hash/hashing_service.php');
+require_once($CFG->dirroot . '/auth/mumie/classes/sso/user/mumie_user.php');
+
+/**
+ * This class provides service functionalities for mumie_users
+ *
+ * @package auth_mumie
+ * @copyright  2017-2023 integral-learning GmbH (https://www.integral-learning.de/)
+ * @author Tobias Goltz (tobias.goltz@integral-learning.de)
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class mumie_user_service {
+    /**
+     * Get a mumie_user instance for a given moodle user and MUMIE Task
+     * @param string    $moodleid
+     * @param \stdClass $mumietask
+     * @return mumie_user
+     */
+    public static function get_user(string $moodleid, \stdClass $mumietask) : mumie_user {
+        if (self::use_id_masking($mumietask)) {
+            $mumieid = hashing_service::generate_hash($moodleid, $mumietask)->get_hash();
+        } else {
+            $mumieid = $moodleid;
+        }
```

Comment:
As an java devolper this is kind of weird. Elvis is more readable for me.
$mumie_id = self::use_id_masking($mumietask)
    ? hashing_service::generate_hash($moodleid, $mumietask)->get_hash()
    : $moodleid;

![image](https://github.com/integral-learning/moodle-auth_mumie/assets/73169661/0d5d38fd-61d5-49b2-ac0f-bb7f71801c51)

Reply:
I usuall don't like ternary operator that are not absolutely trivial. They often decrease readability. But I think it might work here.

Changed